server/src/main/java/org/opensearch/identity/Principals.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.identity;
+
+import java.security.Principal;
+
+/**
+ * Available OpenSearch internal principals
+ *
+ * @opensearch.experimental
+ */
+public enum Principals {
+
+    /**
+     * Represents a principal which has not been authenticated
+     */
+    UNAUTHENTICATED(new StringPrincipal("Unauthenticated"));
```

Comment:
This seems to be very limited set of possible principals, I think we should not model it as an enumeration but constants:

```
public final class BuiltinPrincipals {
    public static final Principal UNAUTHENTICATED = new StringPrincipal("Unauthenticated");
}
```

Reply:
Enums imply closed set of choices - this is not the case here